pandas/core/indexes/base.py

```diff
@@ -2404,6 +2408,28 @@ def is_all_dates(self) -> bool:
         )
         return self._is_all_dates
 
+    @final
+    @classmethod
+    def _is_num_index(cls) -> bool:
+        """
+        Check if this is a NumericIndex, but *not* Int64Index, UInt64Index, FloatIndex.
+
+        Used to check if an operation should return NumericIndex or plain Index.
+        """
+        from pandas.core.indexes.numeric import (
+            Float64Index,
+            Int64Index,
+            NumericIndex,
+            UInt64Index,
+        )
+
+        if not issubclass(cls, NumericIndex):
+            return False
+        elif issubclass(cls, (Int64Index, UInt64Index, Float64Index)):
+            return False
+        else:
+            return True
```

Comment:
Could you check `self.__class__ is NumericIndex` for exact class check instead of isinstance/issubclass ?

Reply:
Ah, I see this is now a classmethod. Why does it need to be a class method instead of an instance property?

Reply:
IMO users should be able to subclass `NumericIndex`. If we do exact checks, that will not be possible.

This method can be revoved once we drop `Int64Index`, so it's just a temporary clutch to maintain backward compat,

Reply:
This method is used in construction methods ( `__new__`) in some case, so having it a classmethod simplifies that a bit.

Reply:
Can you point to an example where it is used in ``__new__`` on a class object? (from quickly skimming the diff, all use cases of this method seem to be using instances)

If being able to subclass `NumericIndex` is the concern (I am personally not sure we should explicitly support it though, but that's another discussion), it could maybe also be an option to have what is now `NumericIndex` as `BaseNumericIndex`, and then `NumericIndex` being a simple shallow subclass from that, and let Int64Index et al also subclass the `BaseNumericIndex` (so changing the subclass hierarchy so that NumericIndex is not a parent of Int64Index et al). Which would mean you can safely do `isinstance(idx, NumericIndex)` to find NumericIndex instances without also catching Int64Index et al.  
(note I didn't check in your diff if this would actually be helpful or rather complicate code handling index objects, but I think I would find it a cleaner way to check for the new-style NumericIndex instances)

Reply:
I've added an atrribute  boolean `_is_numeric_index` instead. That simplifies this a bit.

We need to know when something is a `NumericIndex` but not a `Int64Index`, a `UInt64Index` or `Float64Index`, so we need something like this until pandas 2.0/until we drop `Int64Index` and friends.